libs/api-errors.js

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable max-classes-per-file */
+
+/**
+ * General Error Object, gets other props from Error which include
+ *
+ * name, message, stack
+ */
+export class ErrorResponse extends Error {
```

Comment:
# Nitpick
I understand that this handles error messages for the API but if we can include all that in the response object in the controller, there won't be a need for this.

Reply:
@KaiserPhemi you mean like, i create an error method on the response object, like i did for .json to send json responses??

Reply:
@mernxl 
Typically you'll send an API response like this:
```js
return response.send({
message: "",
data,
})
```

This class appears to only handle error message. What I'm saying is that, why create a class for that when I can easily send the error message as an API response like this: 

```js
return response
  .status(500)
  .send({
     message: "Internal Server Error",
     details,
});
```

Reply:
Ok i see. Maybe i state the reasons for my doing that, you see if they are legit.

Although i know theres a down side of having to deal with the error generation and handling but the reasons for doing that are as follows.

- I wanted to reuse the error messages, status codes, so i don't have to always create new ones which may mismatch.
- Gives a general and continuous structure for all backend errors. Especially when it comes to validation errors.